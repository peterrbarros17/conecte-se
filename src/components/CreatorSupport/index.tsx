import Link from "next/link";
import { BiDonateHeart } from "react-icons/bi";

export default function CreatorSupport() {
  return (
    <article className="border-2 border-white rounded-md p-2 my-4">
      <h2 className="relative -top-6 bg-[var(--white)] text-[var(--grey)] w-56 left-5 rounded-md text-[16px] justify-center flex gap-4 items-center">
        <p>Apoie o criador</p>
        <BiDonateHeart size={20} />
      </h2>
      <div className="flex flex-col justify-center items-center md:items-stretch gap-4 md:flex md:flex-row">
        <p className="w-1/2 bg-white text-black p-4 rounded-md">{`Use code "UPETER-YT" na loja da epic games`}</p>
        <p className="w-1/2 bg-white text-black p-4 rounded-md">
          <Link
            href="https://livepix.gg/upeter"
            target="__blank"
            className="underline"
          >
            Envie uma donate para o streamer
          </Link>
        </p>
      </div>
    </article>
  );
}
