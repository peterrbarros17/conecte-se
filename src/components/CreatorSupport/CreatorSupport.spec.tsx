import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CreatorSupport from ".";

describe("CreatorSupport component", () => {
  //Appearance Tests
  it("should render the paragraphs with the correct text and class", () => {
    render(<CreatorSupport />);

    const firstParagraph = screen.getByText("Apoie o criador");
    expect(firstParagraph).toBeInTheDocument();

    const secondParagraph = screen.getByText(
      'Use code "UPETER-YT" na loja da epic games'
    );
    expect(secondParagraph).toBeInTheDocument();

    const thirdParagraph = screen.getByText("Envie uma donate para o streamer");
    expect(thirdParagraph).toBeInTheDocument();

    const paragraphElements = screen.getAllByRole("paragraph");
    paragraphElements.forEach((p, index) => {
      if (index >= 1) {
        expect(p).toHaveClass("w-1/2 bg-white text-black p-4 rounded-md");
      } else {
        expect(p).not.toHaveClass("w-1/2 bg-white text-black p-4 rounded-md");
      }
    });

    expect(screen.getAllByRole("paragraph")).toMatchSnapshot();
  });
});
