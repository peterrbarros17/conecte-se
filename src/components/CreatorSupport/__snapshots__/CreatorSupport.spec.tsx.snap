// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorSupport component should render the paragraphs with the correct text and class 1`] = `
[
  <p>
    Apoie o criador
  </p>,
  <p
    class="w-1/2 bg-white text-black p-4 rounded-md"
  >
    Use code "UPETER-YT" na loja da epic games
  </p>,
  <p
    class="w-1/2 bg-white text-black p-4 rounded-md"
  >
    <a
      class="underline"
      href="https://livepix.gg/upeter"
      target="__blank"
    >
      Envie uma donate para o streamer
    </a>
  </p>,
]
`;
